lib/pf/dhcp/processor.pm

```diff
@@ -0,0 +1,604 @@
+package pf::dhcp::processor;
+
+=head1 NAME
+
+pf::dhcp::processor
+
+=cut
+
+=head1 DESCRIPTION
+
+Processes DHCP packets
+
+=cut
+
+use strict;
+use warnings;
+
+use Try::Tiny;
+use pf::client;
+use pf::constants;
+use pf::constants::dhcp qw($DEFAULT_LEASE_LENGTH);
+use pf::clustermgmt;
+use pf::config;
+use pf::config::cached;
+use pf::db;
+use pf::firewallsso;
+use pf::inline::custom $INLINE_API_LEVEL;
+use pf::iplog;
+use pf::lookup::node;
+use pf::node;
+use pf::util;
+use pf::config::util;
+use pf::services::util;
+use pf::util::dhcp;
+use List::MoreUtils qw(any);
+use pf::api::jsonrpcclient;
+use NetAddr::IP;
+use pf::SwitchFactory;
+use pf::log;
+
+our $logger = get_logger;
+my $force_update_on_ack = isenabled($Config{network}{force_listener_update_on_ack});
```

Comment:
This should not be a global variable since it will any changes will not be picked later.